integration/tests/access/unstaked_node_test.go

```diff
@@ -48,71 +48,106 @@ func (suite *UnstakedAccessSuite) TearDownTest() {
+	//	if err == nil {
+	//		suite.unstakedReader = reader
+	//		break
+	//	}
+	//	if attempts >= 10 {
+	//		require.NoError(suite.T(), err, "could not subscribe to unstaked ghost (%d attempts)", attempts)
+	//	}
+	//}
+}
 
-	// verification node (unused)
-	verConfig := testnet.NewNodeConfig(flow.RoleVerification, testnet.AsGhost())
-	nodeConfigs = append(nodeConfigs, verConfig)
+func buildNetConfig() testnet.NetworkConfig {
 
-	// collection node (unused)
-	collConfig := testnet.NewNodeConfig(flow.RoleCollection, testnet.AsGhost())
-	nodeConfigs = append(nodeConfigs, collConfig)
+	logOnlyFatal := testnet.WithLogLevel(zerolog.FatalLevel)
 
-	conf := testnet.NewNetworkConfig("unstaked_node_test", nodeConfigs)
-	suite.net = testnet.PrepareFlowNetwork(suite.T(), conf)
+	collectionConfigs := []func(*testnet.NodeConfig){
+		testnet.WithAdditionalFlag("--hotstuff-timeout=12s"),
+		testnet.WithAdditionalFlag("--block-rate-delay=100ms"),
+		logOnlyFatal,
+	}
 
-	// start the network
-	suite.ctx, suite.cancel = context.WithCancel(context.Background())
-	suite.net.Start(suite.ctx)
+	consensusConfigs := append(collectionConfigs,
+		testnet.WithAdditionalFlag(fmt.Sprintf("--required-verification-seal-approvals=%d", 1)),
+		testnet.WithAdditionalFlag(fmt.Sprintf("--required-construction-seal-approvals=%d", 1)),
+		logOnlyFatal,
+	)
 
-	unstakedGhost := suite.net.ContainerByID(suite.unstakedID)
-	client, err := common.GetGhostClient(unstakedGhost)
-	require.NoError(suite.T(), err, "could not get ghost client")
-	suite.unstakedGhost = client
-
-	conGhost := suite.net.ContainerByID(suite.conID)
-	client, err = common.GetGhostClient(conGhost)
-	require.NoError(suite.T(), err, "could not get ghost client")
-	suite.conGhost = client
-
-	for attempts := 0; ; attempts++ {
-		reader, err := suite.unstakedGhost.Subscribe(suite.ctx)
-		if err == nil {
-			suite.unstakedReader = reader
-			break
-		}
-		if attempts >= 10 {
-			require.NoError(suite.T(), err, "could not subscribe to unstaked ghost (%d attempts)", attempts)
-		}
+	net := []testnet.NodeConfig{
+		testnet.NewNodeConfig(flow.RoleCollection, collectionConfigs...),
+		testnet.NewNodeConfig(flow.RoleCollection, collectionConfigs...),
+		testnet.NewNodeConfig(flow.RoleExecution, logOnlyFatal),
+		testnet.NewNodeConfig(flow.RoleExecution, logOnlyFatal),
+		testnet.NewNodeConfig(flow.RoleConsensus, consensusConfigs...),
+		testnet.NewNodeConfig(flow.RoleConsensus, consensusConfigs...),
+		testnet.NewNodeConfig(flow.RoleConsensus, consensusConfigs...),
+		testnet.NewNodeConfig(flow.RoleVerification, logOnlyFatal),
+		testnet.NewNodeConfig(flow.RoleAccess, testnet.AsUnstakedNetworkParticipant(), testnet.WithLogLevel(zerolog.InfoLevel)),
+		testnet.NewNodeConfig(flow.RoleAccess, testnet.AsUnstaked(), testnet.WithLogLevel(zerolog.InfoLevel)),
 	}
+
+	return testnet.NewNetworkConfig("unstaked_node_test", net)
 }
 
 func (suite *UnstakedAccessSuite) TestReceiveBlocks() {
```

Comment:
This test starts all node types and the chain starts building blocks. The test then checks if the unstaked AN is indeed getting blocks and its height is advancing.